include/roboteq_controller.hpp

```diff
@@ -0,0 +1,264 @@
+#pragma once
+
+#include <memory> 
+#include <iostream>
+
+#include "i_comm.hpp"
+
+namespace roboteq {
+
+    class roboteq_controller {
+
+        public :
+        explicit roboteq_controller(std::unique_ptr<i_comm> &&comm);
+        roboteq_controller ( const roboteq_controller & ) = delete;
+        roboteq_controller & operator= ( const roboteq_controller& ) = delete;
+        ~roboteq_controller(){}
+
+        // set methods
+        void SetPosition(long position, uint8_t channel){
```

Comment:
function prototypes should be in the hpp. implementation should be in the cpp

Reply:
I keep getting this error in the .cpp `identifier "_comm" is undefined`